src/fu-util.c

```diff
@@ -3006,22 +3008,31 @@ fu_util_update(FuUtilPrivate *priv, gchar **values, GError **error)
 			continue;
 		}
 
-		if (!fu_util_update_device_with_release(priv, dev, rel, &error_local)) {
-			if (g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_NOTHING_TO_DO)) {
-				g_debug("ignoring %s: %s",
-					fwupd_device_get_id(dev),
-					error_local->message);
-				continue;
-			}
-			g_propagate_error(error, g_steal_pointer(&error_local));
-			return FALSE;
+		ret = fu_util_update_device_with_release(priv, dev, rel, &error_install);
+		if (!ret &&
+		    g_error_matches(error_install, FWUPD_ERROR, FWUPD_ERROR_NOTHING_TO_DO)) {
+			g_debug("ignoring %s: %s",
+				fwupd_device_get_id(dev),
+				error_install->message);
+			continue;
 		}
-
 		fu_util_display_current_message(priv);
 
 		/* send report if we're supposed to */
-		if (!fu_util_maybe_send_reports(priv, rel, error))
+		if (!fu_util_maybe_send_reports(priv, rel, &error_report)) {
+			/* install succeeded, but report failed */
+			if (ret)
+				g_propagate_error(error, g_steal_pointer(&error_report));
+			/* install failed, report failed */
+			else
+				g_warning("%s", error_report->message);
```

Comment:
can we have some {} here pls. I think it would be better to return FALSE as soon as we do g_propagate_error

Reply:
fixed